internal/service/lightsail/database.go

```diff
@@ -0,0 +1,462 @@
+				Computed: true,
+			},
+			"cpu_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"ram_size": {
+				Type:     schema.TypeFloat,
+				Computed: true,
+			},
+			"disk_size": {
+				Type:     schema.TypeFloat,
+				Computed: true,
+			},
+			"master_endpoint_port": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"master_endpoint_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"secondary_availability_zone": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"support_code": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": tftags.TagsSchema(),
+		},
+	}
+}
+
+func ResourceDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).LightsailConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
+
+	req := lightsail.CreateRelationalDatabaseInput{
+		MasterDatabaseName:            aws.String(d.Get("master_database_name").(string)),
+		MasterUsername:                aws.String(d.Get("master_username").(string)),
+		RelationalDatabaseBlueprintId: aws.String(d.Get("blueprint_id").(string)),
+		RelationalDatabaseBundleId:    aws.String(d.Get("bundle_id").(string)),
+		RelationalDatabaseName:        aws.String(d.Get("name").(string)),
+	}
+
+	if v, ok := d.GetOk("availability_zone"); ok {
+		req.AvailabilityZone = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("master_password"); ok {
+		req.MasterUserPassword = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("preferred_backup_window"); ok {
+		req.PreferredBackupWindow = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("preferred_maintenance_window"); ok {
+		req.PreferredMaintenanceWindow = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("publicly_accessible"); ok {
+		req.PubliclyAccessible = aws.Bool(v.(bool))
+	}
+
+	if len(tags) > 0 {
+		req.Tags = Tags(tags.IgnoreAWS())
+	}
+
+	resp, err := conn.CreateRelationalDatabase(&req)
+	if err != nil {
+		return err
+	}
+
+	if len(resp.Operations) == 0 {
+		return fmt.Errorf("No operations found for Create Relational Database request")
+	}
+
+	op := resp.Operations[0]
+	d.SetId(d.Get("name").(string))
+
+	err = waitLightsailOperation(conn, op.Id)
+	if err != nil {
+		return fmt.Errorf("Error waiting for Relational Database (%s) to become ready: %s", d.Id(), err)
+	}
+
+	// Backup Retention is not a value you can pass on creation and defaults to true.
+	// Forcing an update of the value after creation if the backup_retention_enabled value is false.
+	if v := d.Get("backup_retention_enabled"); v == false {
+		log.Printf("[DEBUG] Lightsail Database (%s) backup_retention_enabled setting is false. Updating value.", d.Id())
+		req := lightsail.UpdateRelationalDatabaseInput{
+			ApplyImmediately:       aws.Bool(true),
+			RelationalDatabaseName: aws.String(d.Id()),
```

Comment:
Here you are using `id` attribute, but above on lines 161 and 198 you are using the `name`. I would recommend staying consistent even though both hold the same string value.

Reply:
I went through the code and updated all references to use the `id` attribute.